File: 프로그래머스/레벨 3/프로그래머스  최고의 집합.cpp



#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n, int s) {
    vector<int> answer;
    
    if(n > s){
        answer.push_back(-1);
        return answer;
    }
    
    int div = s / n;
    int remain = s % n;
    for(int i=1;i<=n;i++)
        answer.push_back(div);
    
    for(int i=1;i<=remain;i++)
        answer[n-i] += 1;
    
    
    
    
    return answer;
}

File: 프로그래머스/레벨 3/프로그래머스 최고의 집합.py
def solution(n, s):
    answer = []
    
    if n > s:
        return [-1]
    else:
        arr = []
        div = s // n
        rem = s % n
        # 몫만큼을 넣어주어야 한다.
        # 7을 3개의 자연수로 나누어줄 때
        # 2 2 2를 넣어주고 나머지인 1을 마지막에만 더해줌
        
        for i in range(n):
            arr.append(div)
        for i in range(0,rem):
            arr[n-i-1] += 1
    
    return arr
